### DISCUSSION TITLE
Github Copilot Chat Will Not Install in VS Code Dev Container

### DISCUSSION THREAD

**Discussion by nicholasperkins1992** [community/community#53892] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Github Copilot Chat will not install inside dev container in VS code nightly build 1.17.0-Insdier.

The following warning is shown when viewing the extension in VS Code.
```
This extension is disabled in this workspace because it is defined to run in the Remote Extension Host. Please install the extension in '[DEV Container Name]' to enable.
```

When I click Install in [Dev container name] the following error occurs
```
Can't install release version of 'github.copilot-chat' extension because it has no release version.
```

I have verified I am trying to install the prerelease version.

I have also verified that the extension installs and works when not using a dev container

**Comment by Akash1134** [comment]
Hi @nicholasperkins1992 ,

Copilot Chat is in its public preview and must be joined via a [waitlist](https://github.com/github-copilot/chat_waitlist_signup/join). Please checkout the step-by-step instructions on [how to setup Copilot Chat in VS Code](https://techcommunity.microsoft.com/t5/educator-developer-blog/step-by-step-how-to-setup-copilot-chat-in-vs-code/ba-p/3849227?WT.mc_id=academic-99253-abartolo).

Prerequisites

- Email confirmation from our team that you have been admitted to the waitlist. 
- An active [GitHub account](https://github.com/signup) and latest version of [VSCode](https://code.visualstudio.com/download)
- [GitHub Copilot ](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot)extension (Note that Copilot Nightly has been [deprecated](https://github.com/orgs/community/discussions/59964))
- [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) extension from VSCode

Thanks!

**Reply by nicholasperkins1992** [reply]
Yes I can confirm i have access, it worked correctly for me as of last week, and GitHub copilot chat works correctly for me when using VS Code without a Remote Host Extension. The issue occurs when I attach VS Code to a Dev Container. Only then do i get an error when installing the GitHub Copilot Chat extension.

I did not have this issue until chat was split into its own extension from the github copilot nightly extension

**Reply by nicholasperkins1992** [reply]
<img width="706" alt="image" src="https://user-images.githubusercontent.com/54188188/235228483-2bbee5d9-35fa-4ee7-af52-e0c0a311c37d.png">
I have attached an image of the error i encounter when attempting to install GitHub copilot chat in my dev container

**Reply by ckcollab** [reply]
I'm experiencing this as well, although I disabled the Docker extension so that shouldn't be playing a part:

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/2185159/236949347-1ec79fbd-c36b-4f42-a789-c9a4c3d876c4.png">

I've been accepted into the beta, so I should be able to install this I believe?

**Comment by Marius-Ar** [comment]
I've been experiencing the same problem

As stated in the VS Code extensions explorer, Copilot Chat is a pre-release
To install it click on the down arrow next to "Install" then "Install Pre-Release Version"

**Reply by StanFromTraction** [reply]
Thanks

**Reply by codenjoyme** [reply]
Thanks

**Reply by jsmile** [reply]
Thanks so much.

**Reply by kevin-tournie** [reply]
Thanks !

**Reply by Luis2088** [reply]
You are my benefactor.

**Comment by davidawad** [comment]
Same problem 

```
Can't install release version of 'github.copilot-chat' extension because it has no release version.
```

**Reply by Marius-Ar** [reply]
https://github.com/community/community/discussions/53892#discussioncomment-5869501

**Reply by junaid1460** [reply]
<img width="674" alt="Screenshot 2023-05-16 at 7 46 37 AM" src="https://github.com/community/community/assets/3303745/742a79cf-01e8-4b26-813e-ec7fb891d310">

**Reply by nicholasperkins1992** [reply]
I tried it and it did not work unfortunately.

**Comment by jsborjesson** [comment]
I had the same problem, but eventually got it to work after restarting VSCode.

**Comment by szubia** [comment]
Open Code and run this command with Ctrl-P:
`ext install GitHub.copilot-nightly`

It's the only way I was able to get it to install.

**Comment by jeffscottward** [comment]
I have insider and nightly and the pre-release of chat installed (supposedly even though I don't see it in the sidebar) but nothing is happening

**Comment by hankei6km** [comment]
Hi,

Same problem with using CodeSandbox extension(`CodeSandbox-io.codesandbox-projects`) in VSCode Insiders. 
(CodeSandbox extension is probably using remote-ssh to connect to the server)

```
Can't install release version of 'github.copilot-chat' extension because it has no release version.
```

I asked Copilot Chat for a solution. 
Copilot Chat replied wrokaround with using`remote.extensionKind`.


It doesn't seem like the best way, but it was solved by adding the following to `settings.json`.


```json
  "remote.extensionKind": {
    "github.copilot-chat": "ui"
  }
```
![copilot-chat is installed locally](https://github.com/community/community/assets/1386689/c7754701-3062-473b-97c3-807c02674ccd)

**Reply by hankei6km** [reply]
After upgrading GitHub Copilot Chat, the workaround is no longer needed in my environment.

Name: GitHub Copilot Chat
Id: GitHub.copilot-chat
Description: AI chat features powered by Copilot
Version: 0.1.2023051801

![copilot-chat-installed](https://github.com/community/community/assets/1386689/1e1e4950-cdae-4b1f-8b58-59058e3ff63e)

~However, Dev Container environment still required  a workaround(`remote.extensionKind`).~

After updating to a newer version, it became possible to install to DevContainer.

Name: GitHub Copilot Chat
Id: GitHub.copilot-chat
Description: AI chat features powered by Copilot
Version: 0.1.2023051803

**Comment by bHimes** [comment]
Solution [HERE ](https://github.com/orgs/community/discussions/55019#discussioncomment-5890825) worked, pasted below for convenience HT @pkerschbaum

```bash 
# Run inside Dev container
code-insiders --install-extension github.copilot-chat@prerelease
```

**Reply by serefarikan** [reply]
Thanks! I used this over an ssh connection to a non-docker, actual linux machine and it solved the problem.

**Comment by jeffscottward** [comment]
I finally got it working by just completely redownloading the Insiders app which suddenly had a new color icon.

I think they screwed up the build and released the regular stable as an insider build.

**Comment by stevenlafl** [comment]
Install the Insider in WSL and it will automatically install the Copilot Chat extension. This fooled me at first too when I got the error trying to install it in WSL.

**Comment by Misaka-Tang** [comment]
I can't see the interface in vscode mac after installing the preRelease extension.

**Reply by zakriabilal** [reply]
I am having the same problem, nothing is showing up on MAC, I have tried nightly as well as the main copilot plugin. Github Copilot labs is also installed.

**Reply by fan-k-tamura** [reply]
I had the same issue.
After Run "Chat: Focus on GitHub Copilot View" from command palette (Shift+Ctrl+P in Windows), I was able to see Chat extension.

**Reply by lyqht** [reply]
same here, and i can't find "Chat: Focus on GitHub Copilot View" in command palette.

**Reply by Misaka-Tang** [reply]
Any some solutions?

**Comment by jsmile** [comment]
GitHub Copilot Chat is working well in VSCode - Insider.

But I sucessfully installed GitHub Copilot Chat pre-release, 
I cannot see GitHub Copilot Chat extension icon on the left bar of VSCode.

What is my problem ?

**Comment by llopes-webera** [comment]
Yeah, same here. Even when I clicked in "Join waitlist" it said "You're already in." But I'm not able to use it.

**Comment by colutti** [comment]
Same here on a mac.

**Comment by sikhaman** [comment]
installed pre-release but do not see chat icon

**Comment by jsmile** [comment]
GitHub Copilot Chat is working well only in VSCode - Insider.
Just VSCode says errors.

**Reply by aleleba** [reply]
I have the same problem

**Comment by taurisjoe** [comment]
Is there a way to install the prerelease version automatically in a devcontainer?  I've tried adding '@prerelease' to the extensions section of my devcontainer.json file but the install for Chat and Copilot just hang and never get installed.  I have to load the container with the released version, then go to extensions and specify the prerelease versions for both then reopen the container. 
```
	"customizations": {
		"vscode": {
			"extensions": [
				"github.copilot-chat",    // Can't use @prerelease tag
				"github.copilot"     // Can't use @prerelease tag
			],
```
